### PR TITLE
Release/v0.22.0

### PR DESCRIPTION
**Pull request by ebuchman**
- Merge in go-crypto
- Merge in abci
- Merge in tmlibs
- Prometheus metrics
- Stability fixes
- Various other ...

### CODE REVIEW

**Comment by ebuchman** [comment]
Note we may want to include https://github.com/tendermint/tendermint/pull/1824

**Comment by ebuchman** [comment]
Also need to update the spec
- new hash functions
- Updated the Amino crypto table

File: consensus/state.go [hunk 1]
```diff
@@ -320,10 +320,7 @@ func (cs *ConsensusState) OnStop() {
 
 	cs.timeoutTicker.Stop()
 
-	// Make BaseService.Wait() wait until cs.wal.Wait()
```
Comment (melekes) [inline consensus/state.go]: what is the motivation behind this change?

Comment (ebuchman) [reply consensus/state.go]: SDK tests are failing non-deterministically with "cant open wal". Those tests do a lot of stopping and starting. So I went looking into this and noticed that the code here actually never ran and we never stopped the WAL - unless I'm missing something.
